### ISSUE TITLE
How to view the payload and http header information?

### ISSUE THREAD

**Issue by kkatta** [gravitee-io/issues#477]
I need to debug the payload and headers that is being applied after every policy enforcement across the policy chain. Where can I set the server logs to view the same?

Thanks
Kishore

**Comment by brasseld** [comment]
Hi @kkatta,

The only way to debug after every policy is by using the logging policy. By doing so, logs are written into the `gravitee.log` file.

You can also activate `Dump HTTP requests` in your API's gateway configuration:
<img width="1570" alt="capture d ecran 2017-03-05 a 17 18 38" src="https://cloud.githubusercontent.com/assets/133094/23589110/ccdd9158-01c7-11e7-8006-062b9448cd5d.png">

When enabled, you can have a look to logs into `gravitee-httpdump.log` file.

**Comment by NicolasGeraud** [comment]
hi @kkatta is it ok for you?
Can I close this issue ?

**Comment by kkatta** [comment]
Yes Nicolas we can close the issue. I am able to see the log dump. However, just wondering why the logs are not capturing the http headers?


Thanks

________________________________

**Comment by brasseld** [comment]
When you say:
> However, just wondering why the logs are not capturing the http headers?

You mean the logging policy ? or the log dump ?

**Comment by kkatta** [comment]
Two things:
1. I mean for the logging policy I am not able to see http headers information. 
2. The oauth policy applied will validate the token authenticity which is an API call to the authorization server. I don't see the request/response to/from authorization server getting captured  as part of logging policy and http dumps

**Comment by brasseld** [comment]
> 1. I mean for the logging policy I am not able to see http headers information.

See https://github.com/gravitee-io/issues/issues/481

> 2. The oauth policy applied will validate the token authenticity which is an API call to the authorization server. I don't see the request/response to/from authorization server getting captured as part of logging policy and http dumps

I agree that we must be able to trace HTTP requests / responses to authorization server.
What I'm not sure is that, IMHO, these logs must no be part of the gateway.log and I think it will be more convenient to use a dedicated log file, something like `oauth2-httpdump.log`. WDYT ?
